post/webnote-extension/index.md

```diff
@@ -0,0 +1,658 @@
+    border: 1px solid transparent;
+    padding: 0.6em 1.2em;
+    font-size: 1em;
+    font-weight: 500;
+    font-family: inherit;
+    background-color: #1a1a1a;
+    cursor: pointer;
+    transition: border-color 0.25s;
+}
+button:hover {
+    border-color: #646cff;
+}
+button:focus,
+button:focus-visible {
+    outline: 4px auto -webkit-focus-ring-color;
+}
+```
+
+Role: Its the `customer` role we created. To check, check the url in `https://directus-supabase.onrender.com/admin/settings/roles/1c005600-202d-429c-81bb-452a70aec7e1`
+
+### Load Signup route
+
+-   Open `src/router.js`
+-   Import `signup.vue`
+
+```js
+import SignupView from "../views/signup.vue";
+```
+
+Add the route in `routes`
+
+```js
+   {
+        path: "/signup",
+        name: "signup",
+        meta: { public: true },
+        component: SignupView,
+    },
+```
+
+![Signup](./images/signup.png)
+
+## Setup Login
+
+Create new file `src/views/login.vue`
+
+```html
+<template>
+    <div>
+        <div style="text-align: right">
+            <span
+                @click="$router.push({ name: 'signup' })"
+                style="text-decoration: underline; cursor: pointer"
+            >
+                Signup
+            </span>
+        </div>
+        <form @submit.prevent="login">
+            <label for="email" style="display: block">Email</label>
+            <input type="email" id="email" required v-model="email" />
+
+            <label for="password" style="margin-top: 8px">Password</label>
+            <input type="password" id="password" v-model="password" required />
+
+            <button type="submit" style="display: block; margin-top: 16px">
+                Login
+            </button>
+        </form>
+    </div>
+</template>
+
+<script>
+    export default {
+        inject: ["directus"],
+        data() {
+            return {
+                email: "",
+                password: "",
+            };
+        },
+        methods: {
+            async login() {
+                try {
+                    await this.directus.login(this.email, this.password);
+                    this.$router.push({ name: "home" });
+                } catch ({ errors }) {
+                    console.log("🚀 ~ login ~ errors:", errors);
+                    if (errors[0].extensions.code === "INVALID_CREDENTIALS")
+                        alert("Invalid Email or password");
+                    else
+                        alert(
+                            "Something went wrong. Try again after some time..."
+                        );
+                }
+            },
+        },
+    };
+</script>
+
+<style>
+    input,
+    label {
+        display: block;
+        width: 100%;
+    }
+</style>
+```
+
+### Add login route
+
+```js
+import LoginView from "../views/login.vue";
+```
+
+```js
+  {
+        path: "/login",
+        name: "login",
+        meta: { public: true },
+        component: LoginView,
+    },
+```
+
+![Login](./images/login.png)
```

Comment:
````suggestion
## Login Users

Create new file `src/views/login.vue`:

```html
<template>
    <div>
        <div style="text-align: right">
            <span
                @click="$router.push({ name: 'signup' })"
                style="text-decoration: underline; cursor: pointer"
            >
                Signup
            </span>
        </div>
        <form @submit.prevent="login">
            <label for="email" style="display: block">Email</label>
            <input type="email" id="email" required v-model="email" />

            <label for="password" style="margin-top: 8px">Password</label>
            <input type="password" id="password" v-model="password" required />

            <button type="submit" style="display: block; margin-top: 16px">
                Login
            </button>
        </form>
    </div>
</template>

<script>
    export default {
        inject: ["directus"],
        data() {
            return {
                email: "",
                password: "",
            };
        },
        methods: {
            async login() {
                try {
                    await this.directus.login(this.email, this.password);
                    this.$router.push({ name: "home" });
                } catch ({ errors }) {
                    console.log("🚀 ~ login ~ errors:", errors);
                    if (errors[0].extensions.code === "INVALID_CREDENTIALS")
                        alert("Invalid Email or password");
                    else
                        alert(
                            "Something went wrong. Try again after some time..."
                        );
                }
            },
        },
    };
</script>

<style>
    input,
    label {
        display: block;
        width: 100%;
    }
</style>
```

### Add login route

Open `src/router.js` and import `login.vue`:

```js
import LoginView from "../views/login.vue";
```

Add the route in `routes`:

```js
{
    path: "/login",
    name: "login",
    meta: { public: true },
    component: LoginView,
},
```

![Login](./images/login.png)
````

Reply:
remove styling